### PR TITLE
Provide custom names for tests

### PR DESCRIPTION
**Pull request by kklein**

### CODE REVIEW

**Comment by kklein** [comment]
fyi @jonashaag @YYYasin19

**Comment by jonashaag** [comment]
🎉

File: tests/unit/test_pytest.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+def test_collect_data_tests():
+    req = WithinRequirement.from_raw_query("select * from example", "example_table")
+    req.add_n_rows_equality_constraint(42)
+    test_func = pytest_integration.collect_data_tests([req])
+    (parametrize_mark,) = test_func.pytestmark
+    assert parametrize_mark.name == "parametrize"
+    assert parametrize_mark.args[1] == list(req)
+    assert [parametrize_mark.kwargs["ids"](c) for c in parametrize_mark.args[1]] == [
+        "NRowsEquality::example_table"
+    ]
```
Comment (YYYasin19) [inline tests/unit/test_pytest.py:15]: Is this the same?
```py3
assert [parametrize_mark.kwargs["ids"](req) for req in list(req)] == ["NRowsEquality::example_table"]
```

**Comment by YYYasin19** [comment]
Very nice! 🥳 
I think it solves the use case nicely.

File: tests/unit/test_pytest.py (hunk 1 above)
Comment (kklein) [reply tests/unit/test_pytest.py:15]: I find `req in list(req)` a little confusing. Did you mean `[parametrize_mark.kwargs["ids"](constraint) for constraint in req]`?

If so, it should be the same if everything goes well. Using the indirection via `parametrize_mark.args` might be considered a further test whether that is actually the case.

But maybe I'm missing something.

**Review by ivergara — APPROVED** [review]

**Pull request merged**